crates/sui-benchmark/src/lib.rs

```diff
@@ -416,13 +417,11 @@ impl ValidatorProxy for LocalValidatorAggregatorProxy {
                 Err(e) => tracing::warn!("Failed to submit certificate: {e}"),
             }
 
-            // TODO: We can't stop sending transactions after receiving a quorum of replies otherwise
-            // some validators may not be able to process the next transactions. However this is a
-            // problem because: (i) it biases latency which should be computed upon receiving a quorum
-            // of certificate, and (ii) it prevents us from running benchmarks under (crash-)faults.
-            // if total_stake >= self.committee.quorum_threshold() {
-            //     break;
-            // }
+            if total_stake >= self.committee.quorum_threshold() {
+                // Do not wait on executions at all validators. Instead, rely on Narwhal to keep
+                // validators in sync. 
```

Comment:
Last time I tried narwhal was too slow to sync and most dependent transactions were rejected because of that (which biased the load generator). Should we have a dedicated task that simply waits on the remaining replies while the main task moves to the next transaction after receiving a quorum of replies?

Reply:
Hmm was this a recent experiment or done more than 3 months ago? At `main`, submitted certificates should be buffered in transaction manager if they cannot execute immediately. But definitely there can be a perceived latency issue. Spawning a separate task may have lifetime problems, but I can give that a try and make a bigger refactor.

Reply:
it is a few weeks old, should we go for your change and see if a bigger refactor is needed?

Reply:
Sounds good. I will make a stab at the refactor, test it out in private testnet and merge what works. Hopefully it will be good for other scenarios too.